### PR TITLE
1.8.5 rebase

### PR DESCRIPTION
**Pull request by liggitt**
Based on https://github.com/openshift/kubernetes/commits/release-1.8.5

### CODE REVIEW

File: hack/update-deps.sh [hunk 1]
```diff
@@ -7,6 +7,56 @@ set -o pipefail
...
+    git checkout -b "${KUBE_BRANCH}" "origin/${KUBE_BRANCH}"
+  popd
+
+  # Make a repo for each staging dir
+  for pkg in kubernetes/staging/src/k8s.io/*; do
+    dir=$(basename $pkg)
+    echo === $dir
+    # Check out the corresponding kubernetes/* repo
+    git clone https://github.com/kubernetes/$dir.git $dir || true
+    glide mirror set git@github.com:openshift/kubernetes-$dir file://$mirror/$dir --vcs git
```
Comment (deads2k) [inline hack/update-deps.sh]: does this permanently dirty my glide state?

File: hack/update-deps.sh [hunk 2]
```diff
@@ -7,6 +7,56 @@ set -o pipefail
 cd "$(dirname "${BASH_SOURCE}")/.."
 source "$(dirname "${BASH_SOURCE}")/lib/init.sh"
 
+if [[ -z "${KUBE_BRANCH:-}" ]]; then
```
Comment (deads2k) [inline hack/update-deps.sh]: I'm probably reading this wrong, but does this mean we can no longer run `hack/update-deps.sh` without setting this env var?

File: hack/update-deps.sh [hunk 3]
```diff
@@ -7,6 +7,56 @@ set -o pipefail
...
+    pushd $dir
+      # Create a branch based on the upstream tag
+      git fetch origin --tags
+      git reset --hard
+      git clean -fdx
+      git checkout master
+      git branch -D "${KUBE_BRANCH}" || true
+      git checkout -b "${KUBE_BRANCH}" "${KUBE_TAG}"
+      # Copy current state of carries and commit
+      cp -fr ../$pkg/* ./
```
Comment (deads2k) [inline hack/update-deps.sh]: This seems really weird.  It's building an incorrectly history here so you can vendor it or something?

**Comment by deads2k** [comment]
Remove `Allow updating deps directly from openshift/kubernetes` to be considered separately and it lgtm.

The additions to that script look odd and look to be rebase focused, not focused on the majority of the team using it.  And even as someone trying to rebase, I'm not clear on why I'd do that instead of keeping the forks up to date.

File: hack/update-deps.sh (hunk 1 above)
Comment (mfojtik) [reply hack/update-deps.sh]: The mirrors are stored in a mirrors.yaml file in your GLIDE_HOME.

File: hack/update-deps.sh (hunk 3 above)
Comment (liggitt) [reply hack/update-deps.sh]: we never vendor the repo forks (this script deletes and symlinks the staging dirs at the end). this is purely so glide can find the source for dependency computation.

**Comment by liggitt** [comment]
> Remove Allow updating deps directly from openshift/kubernetes to be considered separately and it lgtm.

The release-1.8.5 branches don't exist in the staging forks, so update-deps.sh wouldn't be runnable without that commit.

> The additions to that script look odd and look to be rebase focused, not focused on the majority of the team using it.

Everyone using this script has to worry about the origin->kubernetes->forks flow or they'll overwrite carries.

> And even as someone trying to rebase, I'm not clear on why I'd do that instead of keeping the forks up to date.

Because the forks have no utility beyond dependency calculation and are massively painful to maintain. We don't actually use them in vendoring in origin and we don't intend for people to vendor them independently apart from origin.

File: hack/update-deps.sh (hunk 2 above)
Comment (liggitt) [reply hack/update-deps.sh]: Would like to base this off the pinned level in glide.yaml, this was just to make sure the approach worked

**Comment by mfojtik** [comment]
@liggitt are you looking at the failed GC unit test?

**Comment by liggitt** [comment]
yes

**Comment by mfojtik** [comment]
@liggitt need to add list and watch

**Comment by liggitt** [comment]
> @liggitt need to add list and watch (metav1.Verbs{"delete", "list", "watch"})

yup, repicked in openshift/kubernetes and updated here. also regenerated swagger docs.

File: hack/update-deps.sh (hunk 1 above)
Comment (deads2k) [reply hack/update-deps.sh]: > The mirrors are stored in a mirrors.yaml file in your GLIDE_HOME.

That sounds like it permanently dirties my glide state.  Globally even, so its not contained to this repo.

**Comment by deads2k** [comment]
> Because the forks have no utility beyond dependency calculation and are massively painful to maintain. We don't actually use them in vendoring in origin and we don't intend for people to vendor them independently apart from origin.

That's not entirely true.  `image-registry` and `oc` have to vendor them in.  When we started, there were a couple other who were trying to vendor openshift/origin too.

**Comment by liggitt** [comment]
this is ready

**Comment by liggitt** [comment]
extended image failure is not caused by this PR (seen in other PRs against master)

extended build failures are inconsistent (one failure in https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/17849/test_pull_request_origin_extended_builds/958/, different failures in https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/17849/test_pull_request_origin_extended_builds/962/)

**Comment by deads2k** [comment]
/retest

If nothing new merges into https://github.com/openshift/origin/commits/release-3.8, I'll green button on green tests.

**Comment by liggitt** [comment]
extended build flake, green otherwise

**Comment by liggitt** [comment]
resynced openshift/kubernetes from release-3.8, rebased (two UPSTREAM commits came in)

**Comment by deads2k** [comment]
Green as it will get.   Merging.

**Pull request merged**